include/marisa/base.h

```diff
@@ -28,14 +29,13 @@ typedef uint32_t marisa_uint32;
 typedef uint64_t marisa_uint64;
 #endif  // _MSC_VER
 
-#if defined(_WIN64) || defined(__amd64__) || defined(__x86_64__) || \
-    defined(__ia64__) || defined(__ppc64__) || defined(__powerpc64__) || \
-    defined(__sparc64__) || defined(__mips64__) || defined(__aarch64__) || \
-    defined(__s390x__)
+#if (ULONG_MAX == 0xffffffffffffffff)
```

Comment:
`long` is only 32-bit on 64-bit windows, so I would suggest `UINTPTR_MAX`. It's in `cstdint`.

https://godbolt.org/z/b3oPe69Ps

https://en.wikipedia.org/wiki/64-bit_computing#64-bit_data_models

Reply:
Anything but listing all known 64 bit architectures on the planet is fine with me.

Reply:
Seems like `UINTPTR_MAX` is also in `stdint.h` according [POSIX](https://pubs.opengroup.org/onlinepubs/9699919799/basedefs/stdint.h.html#tag_13_47_03_02).

Reply:
But seems like windwos does not provide `stdint.h`?

https://github.com/s-yata/marisa-trie/blob/006020c1df76d0d7dc6118dacc22da64da2e35c4/include/marisa/base.h#L4

So maybe we need keep ifdef exceptions for non-standard platforms...